public/app/features/alerting/unified/state/AlertmanagerContext.tsx

```diff
@@ -45,13 +56,26 @@ const AlertmanagerProvider = ({ children, accessType }: Props) => {
   const defaultSource = GRAFANA_RULES_SOURCE_NAME;
 
   // queryParam > localStorage > default
-  const desiredAlertmanager = sourceFromQuery ?? sourceFromStore ?? defaultSource;
+  const desiredAlertmanager = alertmanagerSourceName ?? sourceFromQuery ?? sourceFromStore ?? defaultSource;
   const selectedAlertmanager = isAlertManagerAvailable(availableAlertManagers, desiredAlertmanager)
     ? desiredAlertmanager
     : undefined;
 
+  const selectedAlertmanagerConfig = getAlertmanagerDataSourceByName(selectedAlertmanager)?.jsonData;
+
+  // determine if we're dealing with an Alertmanager data source that supports the ruler API
+  const isGrafanaAlertmanager = selectedAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+  const isAlertmanagerWithConfigAPI = CONFIG_API_ENABLED_ALERTMANAGER_FLAVORS.includes(
+    selectedAlertmanagerConfig?.implementation!
+  );
+
+  const hasConfigurationAPI = isGrafanaAlertmanager || isAlertmanagerWithConfigAPI;
+
   const value: Context = {
     selectedAlertmanager,
+    hasConfigurationAPI,
+    isGrafanaFlavoredAlertmanager: isGrafanaAlertmanager,
```

Comment:
Should we rename it to `isGrafanaAlertmanager`? :)

Reply:
Yep! Must have missed this when I was renaming symbols